### PR TITLE
Adding basics of fsharp

### PR DESCRIPTION
**Pull request by georga399**
Adding some grammar of F#.

### CODE REVIEW

**Comment by georga399** [comment]
There are only antlr4 files. It's not full grammar of F#. Maybe some rules should be changed.

**Comment by kaby76** [comment]
Please read https://github.com/antlr/grammars-v4/wiki#what-is-required-to-submit-a-grammar and add these additional files: desc.xml, examples/* (test files to parse), readme.md.

I have additional comments for your grammar.

File: fsharp/FSharpLexer.g4 [hunk 1]
```diff
@@ -0,0 +1,216 @@
+lexer grammar FSharpLexer;
+
+WHITE_SPACE: [ \t\r\n]+ -> skip ;
+COMMENT:( ('//'.*?([\n]| EOF))  | ('(*' .*? '*)') )-> skip ;
```
Comment (kaby76) [inline fsharp/FSharpLexer.g4:4]: EOF is never used in a lexer rule. Just make the '\n' (or '\r'!!!!) optional.

File: fsharp/FSharpLexer.g4 [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+INT: [0-9]+('uy'|'y'|'s'|'us'|'u'|'L')?;
+
+FLOAT: [0-9]+'.'[0-9]+ ('f'|'m')?;
+
+INTERPOLATIONSIGN: ('%s'|'%d'|'%f'|'%c');
```
Comment (kaby76) [inline fsharp/FSharpLexer.g4:10]: The parentheses are useless. Please remove them.

Once you get a build of the grammar going, you can look at the static tests to see this and other issues that you can fix.

**Pull request closed without merging**